system/include/emscripten/val.h

```diff
@@ -225,29 +225,18 @@ inline void writeGenericWireType(GenericWireType*& cursor, uint64_t wt) {
 
 template<typename T>
 void writeGenericWireType(GenericWireType*& cursor, T* wt) {
-    #if __ILP32__
-    cursor->w[0].p = wt;
-    #else
-    cursor->p = wt;
-    // FIXME: This requires the JS reading code to be audited to be compatible with it.
-    assert(false);
-    abort();
-    #endif
+    uintptr_t short_ptr = reinterpret_cast<uintptr_t>(wt);
+    assert(short_ptr <= UINT32_MAX);
```

Comment:
is this assert going to get compiled out in release builds? Should we maybe make it a "real" error, or is there some other code that will help the user if they happen to end up with a pointer outside this range?

Reply:
Its only going to fire for users of MEMORY64 + embind...  which is hyper experiential config anyway, so I'm not too worried.